### ISSUE TITLE
fine-tune script

### ISSUE THREAD

**Issue by MasoudJB** [bytedance/UI-TARS#39]
Is it possible to provide a finetuning script to help with fine-tuning the model for better grounding on specific use cases?

**Comment by Lauriecando** [comment]
It's a great work!!! I'm also looking forward to fine-tuning training scripts and data formats to do some specific work.

**Comment by JjjFangg** [comment]
Our model is fine-tuned based on Qwen2VL. For further fine-tuning on specific use cases, we recommend referring to the official Qwen2VL training scripts as a reference.

**Comment by MasoudJB** [comment]
@JjjFangg Thank you for the response, I was under the impression that you have slightly changed the Qwen2VL pipeline by introducing special tokens. It will be significantly useful for the community if we get some guidelines about further fine-tuning to make sure that we use this great SOTA work properly.

**Comment by JjjFangg** [comment]
Thank you for your question. We fully adhere to the original vocabulary of Qwen2VL and have not added any special tokens. Additionally, the model architecture is identical to that of Qwen2VL. We hope this clarifies your concern and are happy to assist further if needed.

**Comment by leeaction** [comment]
> Thank you for your question. We fully adhere to the original vocabulary of Qwen2VL and have not added any special tokens. Additionally, the model architecture is identical to that of Qwen2VL. We hope this clarifies your concern and are happy to assist further if needed.

It is possible to provide some fine-tuning dataset samples? include how to set UI item box data, Action space etc...

**Comment by rubentorresbonet** [comment]
+1, some dataset samples would be great to see the expected format. Thanks!

**Comment by JjjFangg** [comment]
This is a sample from the Mind2Web training set. We organize the data using a history of 5 (with a maximum of 5 images), and the coordinate information is normalized to the range [0, 1000].

[
    {
        "role": "system",
        "content": "You are a helpful assistant.",
        "loss_mask": 0
    },
    {
        "role": "user",
        "content": "You are a GUI agent. You are given a task and your action history, with screenshots. You need to perform the next action to complete the task. \n\n## Output Format\n```\nAction: ...\n```\n\n## Action Space\nclick(start_box="<|box_start|>(x1,y1)<|box_end|>")\nselect(start_box="<|box_start|>(x1,y1)<|box_end|>") # To expand a dropdown menu or select an item from it.\ntype(content="", start_box="<|box_start|>(x1,y1)<|box_end|>") # First click on the textbox, and then type the content.\n\n## User Instruction\nBuy a pop rock album CD from the United Kingdom that was released in 2016, is between £15 and £20 and in perfect condition.\n\n    ",
        "loss_mask": 0
    },
    {
        "role": "assistant",
        "content": "Action: click(start_box="<|box_start|>(495,30),(495,30)<|box_end|>")",
        "loss_mask": 0
    },
    {
        "role": "assistant",
        "content": "Action: click(start_box="<|box_start|>(381,153),(381,153)<|box_end|>")",
        "loss_mask": 0
    },
    {
        "role": "assistant",
        "content": "Action: click(start_box="<|box_start|>(66,263),(66,263)<|box_end|>")",
        "loss_mask": 0
    },
    {
        "role": "assistant",
        "content": "Action: click(start_box="<|box_start|>(66,851),(66,851)<|box_end|>")",
        "loss_mask": 0
    },
    {
        "role": "user",
        "content": "<|vision_start|><|image_pad|>...<|image_pad|><|vision_end|>",
        "loss_mask": 0
    },
    {
        "role": "assistant",
        "content": "Action: click(start_box="<|box_start|>(66,638),(66,638)<|box_end|>")",
        "loss_mask": 0
    },
    {
        "role": "user",
        "content": "<|vision_start|><|image_pad|>...<|image_pad|><|vision_end|>",
        "loss_mask": 0
    },
    {
        "role": "assistant",
        "content": "Action: click(start_box="<|box_start|>(66,86),(66,86)<|box_end|>")",
        "loss_mask": 0
    },
    {
        "role": "user",
        "content": "<|vision_start|><|image_pad|>...<|image_pad|><|vision_end|>",
        "loss_mask": 0
    },
    {
        "role": "assistant",
        "content": "Action: click(start_box="<|box_start|>(66,5),(66,5)<|box_end|>")",
        "loss_mask": 0
    },
    {
        "role": "user",
        "content": "<|vision_start|><|image_pad|>...<|image_pad|><|vision_end|>",
        "loss_mask": 0
    },
    {
        "role": "assistant",
        "content": "Action: click(start_box="<|box_start|>(293,218)<|box_end|>")",
        "loss_mask": 1
    }
]

**Comment by nicken** [comment]
Why is the loss_mask set to 1 only for the last assistant, rather than for the assistant after each image? Is it to ensure that the model can see the latest 5 photos during training (or as many as there are if fewer than 5)?
---

**Comment by JjjFangg** [comment]
Yes, your understanding is correct. This is to ensure that there are no more than five images.

**Comment by Code24Man** [comment]
上面的json样本中只有4张截图，看回答中说的5张截图，请解释下~
